src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -190,6 +190,12 @@ internal void CreateLuceneIndexWriter(OpenMode mode = OpenMode.CREATE)
             }
         }
 
+        private void InitializeIndexSearcher()
+        {
+            dirReader = writer != null ? writer.GetReader(applyAllDeletes: true) : DirectoryReader.Open(indexDir);
+            Searcher = new(dirReader);
+        }
+
```

Comment:
@RobertGlobant20 
I opted to initialize the Searcher (and dirReader) after every index modification that I could find.
Do you think I got them all ?

Reply:
yes, I think all of them were covered.